qutebrowser/mainwindow/mainwindow.py

```diff
@@ -127,6 +127,45 @@ def get_target_window():
         return None
 
 
+def _generate_stylesheet_statusbar():
+    flags = [
+        ('private', 'statusbar.private'),
+        ('caret', 'statusbar.caret'),
+        ('caret-selection', 'statusbar.caret.selection'),
+        ('prompt', 'prompts'),
+        ('insert', 'statusbar.insert'),
+        ('command', 'statusbar.command'),
+        ('passthrough', 'statusbar.passthrough'),
+        ('private-command', 'statusbar.command.private'),
+    ]
+    stylesheet = """
+        StatusBar,
+        StatusBar QLabel,
+        StatusBar QLineEdit {
+            font: {{ conf.fonts.statusbar }};
+            color: {{ conf.colors.statusbar.normal.fg }};
+        }
+
+        StatusBar {
+            background-color: {{ conf.colors.statusbar.normal.bg }};
+        }
+    """
+    for flag, option in flags:
+        stylesheet += """
+            StatusBar[color_flags~="%s"],
+            StatusBar[color_flags~="%s"] QLabel,
+            StatusBar[color_flags~="%s"] QLineEdit {
+                color: {{ conf.colors.%s }};
+            }
+
+            StatusBar[color_flags~="%s"] {
+                background-color: {{ conf.colors.%s }};
+            }
+        """ % (flag, flag, flag,  # noqa: S001
```

Comment:
`%` formatting has been [deprecated](https://docs.python.org/release/3.1.5/library/stdtypes.html#old-string-formatting-operations) for a long time

Reply:
Also, with all the repetitions of `flag` I might prefer keyword format arguments.